website/src/auth.tsx

```diff
@@ -1,31 +1,35 @@
-import { HubCallback } from "@aws-amplify/core"
 import {
   AuthenticationDetails,
   CognitoUser,
   CognitoUserPool,
   CognitoUserSession,
 } from "amazon-cognito-identity-js"
-import Amplify, { Auth, Hub } from "aws-amplify"
 import React, { createContext, useContext, useEffect, useState } from "react"
 
 type UserContextType = {
   user: CognitoUser | null
   setUser: (user: CognitoUser | null) => void
+  authenticated: boolean
+  setAuthenticated: (newState: boolean) => void
   operations: {
     loginUser: (username: string, password: string) => void
+    resetPassword: (username: string) => void
+    changePassword: (verificationCode: string, newPassword: string) => void
   }
 }
 
 const UserContext = createContext<UserContextType>({} as UserContextType)
 
-export const userPool = new CognitoUserPool({
+const userPool = new CognitoUserPool({
   UserPoolId: process.env["DAILP_USER_POOL"] ?? "",
   ClientId: process.env["DAILP_USER_POOL_CLIENT"] ?? "",
 })
 
 export const UserProvider = (props: { children: any }) => {
+  const [authenticated, setAuthenticated] = useState(false)
+
   const [user, setUser] = useState<CognitoUser | null>(
-    userPool.getCurrentUser()
+    authenticated ? userPool.getCurrentUser() : null
```

Comment:
This gets rid of persisted logins, I'm not sure how this is the solution?

Reply:
Thanks for the catch!! For some reason, I was worried userPool.getCurrentUser() would return a user even if they were not authenticated (i.e. started their resetting password flow) so I would check for authenticated, but realize it will always be set to false if the component is refreshed. I'm committing some new changes to fix this.